src/Altinn.Common/Altinn.EFormidlingClient/Altinn.EFormidlingClient/IEFormidlingClient.cs

```diff
@@ -0,0 +1,106 @@
+using System.IO;
+using System.Threading.Tasks;
+using Altinn.Common.EFormidlingClient.Models;
+using Altinn.Common.EFormidlingClient.Models.SBD;
+
+namespace Altinn.Common.EFormidlingClient
+{
+    /// <summary>
+    /// Interface for actions related to the eFormidling Integration Point(IP) API.
+    /// Ref: https://docs.digdir.no/eformidling_nm_restdocs.html
+    /// </summary>
+    public interface IEFormidlingClient
+    {
+        /// <summary>
+        /// Subscribes to IP API with callback URL allowing to get push notifcation for message status
+        /// </summary>
+        /// <param name="name"> Name of the subscription to be created.</param>
+        /// <param name="pushEndpoint"> Callback URL the IP will post status to.</param>
+        /// <param name="resource">Type of subscription e.g. 'messages'.</param>
+        /// <param name="event">Type of event e.g. 'status' </param>
+        /// <param name="filter">Filter parameters for status e.g. 'status=INNKOMMENDE_MOTTAT&direction=INCOMING'</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        Task<bool> SubscribeeFormidling(string name, string pushEndpoint, string resource, string @event, string filter);
+
+        /// <summary>
+        /// Deletes subscription by id
+        /// </summary>
+        /// <param name="id"> Id of previously created subscription</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        Task<bool> UnSubscribeeFormidling(int id);
+
+        /// <summary>
+        /// Creates a message using the Standard Business Document Header specification.
+        /// The eFormidling IP uses this document to route it to the correct receiver(s).
+        /// After creation a conversation is created in the IP keepting track of the messages.
+        /// Ref: https://www.gs1.org/standards/edi/standard-business-document-header-sbdh
+        /// </summary>
+        /// <param name="sbd"> Client provides a SBD DTO popluated with necessary fields </param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        Task<StandardBusinessDocument> CreateMessage(StandardBusinessDocument sbd);
+
+        /// <summary>
+        /// Posts attachments related to the message e.g. binary files, arkivmelding.xml
+        /// </summary>
+        /// <param name="stream">Stream of file content</param>
+        /// <param name="id">Descriptor which contains reference information which uniquely identifies this instance of the SBD between the sender and the receiver.</param>
```

Comment:
A lot of words that don't really say a lot.. consider renaming from id to "sbdId" or the type of id we are actually working with.